### PR TITLE
[FIX] Save user password and email from My Account

### LINKED ISSUE #19604: QA-Regression: User is not able to change the account password.
### Description:

User is not able to change the account password.

### Steps to reproduce:

1. Login to Rocketchat
2. Click on the user icon in the side menu.
3. Click "My Account"
4. Go to Profile and Enter new password in the password field.
5. Enter same password in the confirm password field.
6. Click Save changes button
Issue: Invalid password error is displayed and user is not able to change the password."

### Expected behavior:

Password should be changed successfully.

### Actual behavior:

Error is displayed and password is not changed.
image

### Server Setup Information:

- Version of Rocket.Chat Server: 3.8.0
- Operating System: 
- Deployment Method: 
- Number of Running Instances: 
- DB Replicaset Oplog: 
- NodeJS Version: 
- MongoDB Version:

### Client Setup Information

- Desktop App or Browser Version:
- Operating System:

### PR DESCRIPTION
**Pull request by sampaiodiego**
## Issue(s)

Closes #19604

### CODE REVIEW

**Comment by rasos** [comment]
Great to see this fix, as it is really security related, if users can not change their passwords. Also other profile settings (email, avatar image etc) could not be changed.

File: server/methods/saveUserProfile.js [hunk 1]
```diff
@@ -9,108 +9,108 @@ import { twoFactorRequired } from '../../app/2fa/server/twoFactorRequired';
...
+	}
+
+	if (settings.email) {
+		Meteor.call('setEmail', settings.email);
+	}
+
+	const canChangePasswordForOAuth = rcSettings.get('Accounts_AllowPasswordChangeForOAuthUsers');
+	if (canChangePasswordForOAuth || user.services?.password) {
+		// Should be the last check to prevent error when trying to check password for users without password
+		if (settings.newPassword && rcSettings.get('Accounts_AllowPasswordChange') === true) {
```
Comment (KevLehman) [inline server/methods/saveUserProfile.js:70]: We can remove the `=== true` part of this

Comment (sampaiodiego) [reply server/methods/saveUserProfile.js:70]: yep, this is an old code that was just re-indented, but I agree :+1:

File: app/2fa/server/code/index.ts [hunk 2]
```diff
@@ -148,7 +158,7 @@ function _checkCodeForUser({ user, code, method, options = {}, connection }: ICh
 	let selectedMethod = getMethodByNameOrFirstActiveForUser(user, method);
 
 	if (!selectedMethod) {
-		if (options.disablePasswordFallback || !passwordCheckFallback.isEnabled(user)) {
+		if (!options.requireSecondFactor && (options.disablePasswordFallback || !passwordCheckFallback.isEnabled(user, !!options.requireSecondFactor))) {
```
[the remark this answers is not in the document]
Comment (KevLehman) [reply app/2fa/server/code/index.ts:161]: After offline discussion, this condition makes sense.

**Review by Ishrat-Manzoor — DISMISSED** [review]

**Review by ggazzo — APPROVED** [review]

**Pull request merged**